src/Project.js

```diff
@@ -633,6 +637,19 @@ class Project extends DirItem {
         } else {
             exitValue = this.resultStats.errors > 0 ? 2 : ((this.resultStats.warnings > 0) ? 1 : 0);
         }
+
+        if (this.options.opt.output) {
+            if (typeof this.formatter.writeSummaryToFile === 'function') {
+                let summary = this.formatter.writeSummaryToFile(this.project.name, totalTime, this.fileStats, this.resultStats, score) ;
+                resultSet = summary+resultSet;
+            }
+            if (typeof this.formatter.completeFile === 'function') {
+                resultFull = this.formatter.completeFile(resultSet);
+            }
+            let fileName = this.options.opt.output || "result.txt";
```

Comment:
Maybe check if the dir to the output file exists first. If not, use mkdirp to create it first.

Also, let's check if the output file is named earlier while checking the parsed options instead of providing a default here.

Reply:
I updated it